tutorials/source-modeling/plot_freesurfer_coordinate_frames.py

```diff
@@ -0,0 +1,201 @@
+# hood, so let's do it manually so we understand what is happening. First let's
+# get our data as a 3D array and note that it's already a standard size:
+
+data = np.asarray(t1.dataobj)
+print(data.shape)
+
+###############################################################################
+# These data are voxel intensity values (unsigned integers in the range 0-255).
+# Their positions in the voxel array are related to real-world ``(x, y, z)``
+# positions (in RAS orientation) by the image's affine transformation (noting
+# here that :mod:`nibabel` processes everything in units of millimeters):
+
+print(t1.affine)
+
+###############################################################################
+# .. sidebar:: Under the hood
+#
+#     ``mne.transforms.apply_trans`` effectively does a matrix multiplication
+#     (i.e., :func:`numpy.dot`), with a little extra work to handle the shape
+#     mismatch (the affine has shape ``(4, 4)`` because it includes a
+#     *translation*, which is applied separately).
+#
+# This allows us to take an arbitrary voxel or slice of data and know where it
+# is in real-world ``(x, y, z)``, by applying the affine transformation to the
+# voxel coordinates.
+
+vox = np.array([122, 119, 102])
+xyz_ras = apply_trans(t1.affine, vox)
+print('Our voxel has real-world coordinates {}, {}, {} (mm)'
+      .format(*np.round(xyz_ras, 3)))
+
+###############################################################################
+# If you have a point ``(x, y, z)`` in RAS space and you want the
+# corresponding voxel number, you can get it using the inverse of the
+# affine. This involves some rounding, so it's possible to end up off by one
+# voxel if you're not careful:
+
+ras_coords_mm = np.array([1, -17, -18])
+inv_affine = np.linalg.inv(t1.affine)
+i_, j_, k_ = np.round(apply_trans(inv_affine, ras_coords_mm)).astype(int)
+print('Our real-world coordinates correspond to voxel ({}, {}, {})'
+      .format(i_, j_, k_))
+
+###############################################################################
+# Let's write a short function to visualize where our voxel lies in a saggital
+# image, and annotate it in RAS space (rounded to the nearest millimeter):
+
+
+def imshow_mri(data, vox, xyz, suptitle):
+    """Show an MRI slice with a voxel annotated."""
+    i, j, k = vox
+    fig, ax = plt.subplots(1, figsize=(5, 5))
+    ax.imshow(data[i], vmin=10, vmax=120, cmap='gray', origin='lower')
+    ax.axvline(k, color='y')
+    ax.axhline(j, color='y')
+    for kind, coords in xyz.items():
+        annotation = ('{}: {}, {}, {} mm ⬊'
+                      .format(kind, *np.round(coords).astype(int)))
+        text = ax.text(k, j, annotation, va='baseline', ha='right',
+                       color=(1, 1, 0.7))
+        text.set_path_effects([
+            path_effects.Stroke(linewidth=2, foreground='black'),
+            path_effects.Normal()])
+    # reorient for our data orientation and label
```

Comment:
I think this needs more explanation. Why and how did you decide to reorient? Maybe one should add examples of

- `nib.io_orientation()`
- `nib.orientations.aff2axcodes()`
- `nib.as_closest_canonical()`

before moving on to this plotting example?

Reply:
The logic here is specific to our known orientation, a variant of the general logic lives in https://github.com/mne-tools/mne-python/pull/7725. I agree this should be updated to use it